### ISSUE TITLE
torch.argmin behaves differently on CPU and GPU

### ISSUE THREAD

**Issue by carefree0910** [pytorch/pytorch#14147]
## 🐛 Bug

torch.argmin behaves differently on cpu (which is correct) and gpu (which is wrong)

I'm not sure whether this 'bug' (or 'feature') is expected or not...

## To Reproduce

import torch
base = torch.randint(0, 2, (128, 100)).type(torch.LongTensor).cuda()
print(torch.argmin(base, dim=1))           # Produce wrong answer
print(torch.argmin(base.cpu(), dim=1))  # Produce correct answer

## Expected behavior

GPU behavior should be the same as CPU behavior

## Environment

PyTorch version: 0.4.1
Is debug build: No
CUDA used to build PyTorch: 9.0.176
OS: Ubuntu 16.04 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
CMake version: Could not collect
Python version: 3.5
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration: 
GPU 0: GeForce GTX 1080 Ti
GPU 1: GeForce GTX 1080 Ti
GPU 2: GeForce GTX 1080 Ti
GPU 3: GeForce GTX 1080 Ti
GPU 4: GeForce GTX 1080 Ti
GPU 5: GeForce GTX 1080 Ti
GPU 6: GeForce GTX 1080 Ti
GPU 7: GeForce GTX 1080 Ti
Nvidia driver version: 390.77
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_static_v7.a
Versions of relevant libraries:
[pip] Could not collect
[conda] Could not collect

## Additional context

torch.argmax will also behave differently, and maybe other functions as well

**Comment by SsnL** [comment]
What do you mean that it is wrong? It likely just give different answers because there are more than one correct answer.

**Comment by carefree0910** [comment]
Yes, it is giving different answers, and I really don't know whether it is expected or not... In my opinion, these two statements should return the same result, but it appears that the GPU version returns [16,  0, 32,  ..., 16] and the CPU version returns [1, 0, 3, ..., 1] (which is the exact argmin that I want). I'm new to pytorch so I'm quite confused about this behavior.

**Comment by soumith** [comment]
@carefree0910 argmin returns the index of the minimum value in the dimension. It doesn't have a guarantee to return the index of the **first** minimum. The GPU result is also correct from what `argmin` is supposed to return
